core/core-spacemacs.el

```diff
@@ -82,6 +82,9 @@ the final step of executing code in `emacs-startup-hook'.")
   (setq dotspacemacs-editing-style (dotspacemacs//read-editing-style-config
                                     dotspacemacs-editing-style))
   (configuration-layer/initialize)
+  ;; frame title init
+  (when (and (display-graphic-p) dotspacemacs-frame-title-format)
+    (setq frame-title-format '((:eval (spacemacs/frame-title-prepare)))))
```

Comment:
I would assume the **icon-title-format** should be defined here too. Or separate variable should be introduced that will control **icon-title-format**.